lib/trino-parquet/src/main/java/io/trino/parquet/reader/CalendarUtils.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+import static java.time.ZoneOffset.UTC;
+
+public final class CalendarUtils
+{
+    // https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar#Difference_between_Julian_and_proleptic_Gregorian_calendar_dates
+    private static final int[] JULIAN_GREGORIAN_DIFFS = {2, 1, 0, -1, -2, -3, -4, -5, -6, -7, -8, -9, -10, 0};
+    private static final int[] JULIAN_GREGORIAN_DIFF_SWITCH_DAY = {
+            -719164, -682945, -646420, -609895, -536845,
+            -500320, -463795, -390745, -354220, -317695,
+            -244645, -208120, -171595, -141427
+    };
```

Comment:
Can we build a RangeMap and maintain it ? It would allow us to identify on which set of ranges which dates should we consider ?